joblib/numpy_pickle.py

```diff
@@ -384,29 +398,62 @@ def dump(value, filename, compress=0, cache_size=100, protocol=None):
     using compression can significantly slow down loading. In
     addition, compressed files take extra extra memory during
     dump and load.
+
     """
     if compress is True:
-        # By default, if compress is enabled, we want to be using 3 by
-        # default
+        # By default, if compress is enabled, we want to be using 3 by default
         compress = 3
+    elif isinstance(compress, tuple):
+        # a 2-uple was set in compress
+        if len(compress) != 2:
+            raise ValueError(
+                'Compress argument tuple should contain exactly 2 elements: '
+                '(compress method, compress level)')
+        compress_method = compress[0]
+        compress = compress[1]
+
+    if compress is not False and compress not in range(10):
+        # Raising an error if a non valid compress level is given.
+        raise ValueError(
+            'Non valid compress level given: "{0}"'.format(compress))
+
+    if compress_method not in _COMPRESSORS:
+        # Raising an error if an unsupported compression method is given.
+        raise ValueError(
+            'Non valid compression method given: "{0}"'.format(compress_method))
+
+    if not PY3_OR_LATER and compress_method in ('lzma', 'xz'):
+        import sys
+        raise NotImplementedError("{0} compression are available for python "
+                                  " version >= 3.3. You are using {1}."
+                                  .format(compress_method, sys.version))
```

Comment:
sys.version is very verbose. In my case:
```
3.4.4 |Continuum Analytics, Inc.| (default, Jan 11 2016, 13:54:01) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
```

I am wondering if we could use some simpler version string e.g. something like 
```python
"You are using {1[0]}.{1[1]}.".format(compress_method, sys.version_info)
```